servers/src/main/java/tachyon/worker/file/FileDataManager.java

```diff
@@ -153,6 +153,50 @@ private synchronized boolean fileExistsInUfs(long fileId) throws IOException {
   }
 
   /**
+   * Locks all the blocks of a given file Id.
+   *
+   * @param fileId the id of the file
+   * @param blockIds the ids of the file's blocks
+   * @throws IOException when an I/O exception occurs
+   */
+  public void lockBlocks(long fileId, List<Long> blockIds) throws IOException {
+    Map<Long, Long> blockIdToLockId = Maps.newHashMap();
+    List<Throwable> errors = new ArrayList<Throwable>();
+    synchronized (mLock) {
+      if (mPersistingInProgressFiles.containsKey(fileId)) {
+        throw new IOException("the file " + fileId + " is already being persisted");
+      }
+      try {
+        // lock all the blocks to prevent any eviction
+        for (long blockId : blockIds) {
+          long lockId = mBlockDataManager.lockBlock(Sessions.CHECKPOINT_SESSION_ID, blockId);
+          blockIdToLockId.put(blockId, lockId);
+        }
+      } catch (BlockDoesNotExistException e) {
+        errors.add(e);
+        // make sure all the locks are released
```

Comment:
No, this function is for locking. And the unlock will be done after persist call.